builtin/logical/aws/secret_access_keys_test.go

```diff
@@ -47,53 +48,58 @@ func TestNormalizeDisplayName_NormNotRequired(t *testing.T) {
 }
 
 func TestGenUsername(t *testing.T) {
+	type testCase struct {
+		name             string
+		policy           string
+		userType         string
+		UsernameTemplate string
+		warningExpected  bool
+	}
+
+	tests := map[string]testCase{
+		`^vault-name1-policy1-[0-9]+-[a-zA-Z0-9]+`: {
+			name:             "name1",
+			policy:           "policy1",
+			userType:         "iam_user",
+			UsernameTemplate: `{{ printf "vault-%s-%s-%s-%s" (.DisplayName) (.PolicyName) (unix_time) (random 20) | truncate 64 }}`,
+			warningExpected:  false,
+		},
+		`this---is---a---very---long---name-long------policy------name-[0-9][0-9]`: {
+			name:             "this---is---a---very---long---name",
+			policy:           "long------policy------name",
+			userType:         "iam_user",
+			UsernameTemplate: `{{ printf "%s-%s-%s-%s" (.DisplayName) (.PolicyName) (unix_time) (random 20) }}`,
+			warningExpected:  true,
+		},
+	}
+
+	for k, v := range tests {
+		testUsername, warning, err := genUsername(v.name, v.policy, v.userType, v.UsernameTemplate)
+		if err != nil {
+			t.Fatalf(
```

Comment:
nit: Can the `t.Fatalf` calls be a single line instead of multi-line? It's much more common to have it on a single line unless the line is particularly long

Reply:
nit: I should also mention that you can instead use `require.Error(t, err)`: https://pkg.go.dev/github.com/stretchr/testify/require?utm_source=godoc#Error